code/src/sixsq/nuvla/server/resources/voucher.clj

```diff
@@ -0,0 +1,224 @@
+(ns sixsq.nuvla.server.resources.voucher
+  "
+This resource contains the structure for a voucher, which
+is to be issued by a third party and used by any Nuvla user.
+"
+  (:require
+    [sixsq.nuvla.auth.acl-resource :as a]
+    [sixsq.nuvla.auth.utils :as auth]
+    [sixsq.nuvla.db.impl :as db]
+    [sixsq.nuvla.server.resources.common.crud :as crud]
+    [sixsq.nuvla.server.resources.common.schema :as sc]
+    [sixsq.nuvla.server.resources.common.std-crud :as std-crud]
+    [sixsq.nuvla.server.resources.common.utils :as u]
+    [sixsq.nuvla.server.resources.resource-metadata :as md]
+    [sixsq.nuvla.server.resources.spec.voucher :as voucher]
+    [sixsq.nuvla.server.util.response :as r]
+    [clj-time.core :as time]
+    [sixsq.nuvla.server.util.metadata :as gen-md]
+    [clojure.pprint :as pp]))
+
+
+(def ^:const resource-type (u/ns->type *ns*))
+
+
+(def ^:const collection-type (u/ns->collection-type *ns*))
+
+
+(def collection-acl {:query ["group/nuvla-user"]
```

Comment:
Allowing any user to create vouchers seems potentially dangerous.  Don't know if the collection "add" should be more limited.

Reply:
well, in the end the vouchers will be created by some OCRE entity, which is not defined yet, but it will be a Nuvla user. I'd just leave like that for now and change later once we understand a bit better the workflow. 

All we know is that we (SixSq) are not the ones creating the vouchers